sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -824,8 +824,10 @@ bool TryGetStreamLength(Stream stream, out long length)
 
             // Upload the entire stream
             Task<Response<BlobContentInfo>> UploadStreamAsync()
-                =>
-                client.UploadInternal(
+            {
+                (content, metadata) = TransformContent(content, metadata);
```

Comment:
How does this work with retry policies?  Will we potentially transform the content more than once?

Reply:
I have not tested with forced retries. This is still within the convenience layer, though, so we can't end up calling this method more than once.

The implementation we use is not seekable, so we will throw before we attempt to transform twice. Will run tests and follow up on this.